tests/test_mapper_inheritance.py

```diff
@@ -171,3 +174,123 @@ def test_entity_inherit_none(self):
+
+    def test_extract_resource_requirements_from_tool_multiple(self):
+        cores_req = ResourceRequirement("4", "cores_min")
+        mem_req = ResourceRequirement("16384", "ram_min")
+        gpu_req = ResourceRequirement("1", "cuda_device_count_min")
+
+        tool = mock_galaxy.Tool("test_tool", resource_requirements=[cores_req, mem_req, gpu_req])
+        result = extract_resource_requirements_from_tool(tool)
+        expected = {"cores": 4, "mem": 16384, "gpus": 1}
+        self.assertEqual(result, expected)
+
+    def test_extract_resource_requirements_with_not_implemented_error(self):
+        req = ResourceRequirement("$(4 * 4)", "cores_min")
+
+        tool = mock_galaxy.Tool("test_tool", resource_requirements=[req])
+        result = extract_resource_requirements_from_tool(tool)
+        self.assertEqual(result, {})
+
+    def test_tool_with_resource_requirements_mapping(self):
+        cores_req = ResourceRequirement("4", "cores_min")
+        mem_req = ResourceRequirement("8192", "ram_min")
+
+        tool = mock_galaxy.Tool("test_tool_with_resources", resource_requirements=[cores_req, mem_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_uuid_prefix_for_dynamic_tools(self):
+        tool = mock_galaxy.Tool("test_tool", dynamic_tool=mock_galaxy.DynamicTool("12345-67890-abcdef"))
+        tool.tool_type = "interactive"
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_without_uuid_uses_regular_id(self):
+        tool = mock_galaxy.Tool("hisat")
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertEqual(destination.id, "local")
+
+    def test_tool_provided_resource_merging_with_config(self):
+        cores_req = ResourceRequirement("8", "cores_min")
+        mem_req = ResourceRequirement("16384", "ram_min")
+
+        tool = mock_galaxy.Tool("trinity", resource_requirements=[cores_req, mem_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+        self.assertEqual(destination.id, "k8s_environment")
```

Comment:
It would be good to have an assertion here of destination.params, i.e. `self.assertEqual("--cores=8 --mem=250", destination.params.get("submit_native_specification”))`. There must be a specific expectation of what cores and mem would be in this case.

Reply:
k8s_environment specifies max_accepted_mem 64 so if the resource merging didn't work you should get a job mapper exception and the test fails. I've asserted the concrete values below now